### DISCUSSION TITLE
How to have a custom class name for html tag or body tag for every next js page ?

### DISCUSSION THREAD

**Discussion by 1mehdifaraji** [tailwindlabs/tailwindcss#13260] [Help] [answered]
In `global.css` file I would apply some css classes to `body` and `html` tags, now I need to have a separate class name for a separate page . How can I have a custom `html` or `body` css classname in other next js pages ?

`global.css`
```
@layer base {
  html {
    font-weight: 400;
    @apply font-normal scroll-smooth;
  }
  body {
    @apply bg-dynamicBlack text-neoGray;
  }
}
```

**Comment by Vijaykv5** [comment]
Hi @1mehdifaraji 
To achieve this in Next.js, you can utilize the className prop in your page components. Simply add a unique class name to the html or body element within the JSX of your page component. This will override the styles defined in your global CSS file for that specific page.

**Reply by 1mehdifaraji** [reply]
Can you please provide an example ? Thanks in advance .

**Comment by Vijaykv5** [comment]
if you add html and body tag for styling in global.css, it is applied to whole app.
So what I was saying is that make custom class names for other pages.

```
function MyPage() {
  return (
    <div>
      <html className="custom-html-class">
       
      </html>
      <body className="custom-body-class">
        
      </body>
    </div>
  );
}

export default MyPage;

```

**Reply by 1mehdifaraji** [reply]
This is the structure of my next js app:

![Screenshot 1402-12-26 at 20 26 57](https://github.com/tailwindlabs/tailwindcss/assets/63982703/5bf5a387-30ad-425d-933f-544735b922f8)
![Screenshot 1402-12-26 at 20 31 13](https://github.com/tailwindlabs/tailwindcss/assets/63982703/739a1782-968e-46c9-8859-542ef920bc63)
![Screenshot 1402-12-26 at 20 31 09](https://github.com/tailwindlabs/tailwindcss/assets/63982703/1f2a02b7-caf6-4baa-b33c-1d54d4059483)

I import a global css file in `_app.tsx` like this:

```
import "@/styles/globals.css";
```

My question is how can I apply a tailwindcss class name in about page which is a react component basically and I can't import a css file in about page because I get the following error:

```
Global CSS cannot be imported from files other than your Custom <App>. Due to the Global nature of stylesheets, and to avoid conflicts, Please move all first-party global CSS imports to pages/_app.js. 
```

**Reply by 1mehdifaraji** [reply]
Also If I add a `html` tag in my custom page which is `about.tsx` I would have two `html` tags:

![Screenshot 1402-12-26 at 20 36 31](https://github.com/tailwindlabs/tailwindcss/assets/63982703/58192276-eeb8-430d-b894-b731687e9868)

**Comment by Vijaykv5** [comment, marked as answer]
Ok from my understanding, I assume that you need to have separate styles for your whole app right?
The styles defined in your global CSS file imported in `_app.tsx` will be available throughout your application, including the `about.tsx` page component.

 One way to fix this, you can define different styles on different routes and all the style can be written in global.css. This [link](https://stackoverflow.com/questions/49609211/next-js-custom-class-on-body-using-document-js) may help.

**Reply by 1mehdifaraji** [reply]
I wish tailwindcss itself could've handled this situation to apply certain class name to an element based on a custom route. Thanks for the answer.
